File: business/src/main/java/in/okcredit/merchant/store/database/DateTimeRoomCodec.java
package in.okcredit.merchant.store.database;

import androidx.room.TypeConverter;
import in.okcredit.merchant.utils.DateTimeMapper;
import org.joda.time.DateTime;

public final class DateTimeRoomCodec {
    @TypeConverter
    public static DateTime fromEpoch(long timestamp) {
        return DateTimeMapper.fromEpoch(timestamp);
    }

    @TypeConverter
    public static long toEpoch(DateTime dateTime) {
        return DateTimeMapper.toEpoch(dateTime);
    }

    private DateTimeRoomCodec() {}
}

File: rewards/src/main/java/in/okcredit/merchant/rewards/store/database/DateTimeRoomCodec.kt
package `in`.okcredit.merchant.rewards.store.database

import `in`.okcredit.merchant.rewards.utils.DateTimeMapper
import androidx.room.TypeConverter
import org.joda.time.DateTime

object DateTimeRoomCodec {
    @JvmStatic
    @TypeConverter
    fun fromEpoch(timestamp: Long): DateTime = DateTimeMapper.fromEpoch(timestamp)

    @JvmStatic
    @TypeConverter
    fun toEpoch(dateTime: DateTime?): Long = DateTimeMapper.toEpoch(dateTime)
}
